pyiceberg/table/__init__.py

```diff
@@ -153,7 +155,21 @@
 ALWAYS_TRUE = AlwaysTrue()
 TABLE_ROOT_ID = -1
 
-_JAVA_LONG_MAX = 9223372036854775807
+
+def _apply_logical_conversion(table_schema: Schema, task_schema: Schema) -> Schema:
```

Comment:
We can defer this to a later PR, but just want to put it out here.

In PyIceberg we have a little bit an obsession with the visitor pattern. Using the `SchemaWithPartnerVisitor` you can traverse two schema's at once. An example is the `ArrowProjectionVisitor` and can be found in `pyarrow.py`. The `ArrowAccessor` does the lookups by ID, but I think here name makes more sense since we don't have the IDs (yet).